### ISSUE TITLE
Linked containers update failed

### ISSUE THREAD

**Issue by Sveeeeeen** [containrrr/watchtower#671]
**Describe the bug**
Linked containers are not started in the right order.

**To Reproduce**
I have a mariadb docker container and a gitea which is linked to the database container.
watchtower is stopping the database first then gitea. But watchtower is trying to recreate gitea first which fails because database is still stopped.

**Expected behavior**
Recreate the linked container first.

**Logs from running watchtower with the `--debug` [option**]
```
2020-11-05T20:38:01+01 [I] Removing image sha256:c4655f911514fc4403c128227686bb908f858e0575e0e12368ed2a52d4a4a43d
2020-11-05T20:38:00+01 [I] Removing image sha256:ff8a9d7eb7262169e85e1c3d385c8d4817a9bdebb4f816e0c6ebed8014245d05
2020-11-05T20:37:59+01 [D] Starting container /database (eeefc06124d6b959b9c8565f50ad1f1d7a1c902a50df44d23b3a2c6acc603c2b)
2020-11-05T20:37:59+01 [I] Creating /database
2020-11-05T20:37:59+01 [E] Error response from daemon: could not get container for database: No such container: database
2020-11-05T20:37:58+01 [I] Creating /gitea
2020-11-05T20:37:56+01 [D] Removing container 4aa7c13bfe4208a3de2c8b4ba723447bd402596cd2f47e6e7ce42becdf2d015d
2020-11-05T20:37:55+01 [I] Stopping /gitea (4aa7c13bfe4208a3de2c8b4ba723447bd402596cd2f47e6e7ce42becdf2d015d) with SIGTERM
2020-11-05T20:37:50+01 [D] Removing container 51b013ad5038f674a932ca3989f0eb0f707610afbc4d6e0169a9c4d5fd0d0044
2020-11-05T20:37:46+01 [I] Stopping /database (51b013ad5038f674a932ca3989f0eb0f707610afbc4d6e0169a9c4d5fd0d0044) with SIGTERM
2020-11-05T20:37:46+01 [D] No new images found for /portainer_ce
2020-11-05T20:37:44+01 [D] Got image name: portainer/portainer-ce:latest
2020-11-05T20:37:44+01 [D] No credentials for portainer in /config.json
2020-11-05T20:37:44+01 [D] Pulling portainer/portainer-ce:latest for /portainer_ce
2020-11-05T20:37:44+01 [D] No new images found for /phpmyadmin
2020-11-05T20:37:43+01 [D] Got image name: phpmyadmin/phpmyadmin:latest
2020-11-05T20:37:43+01 [D] No credentials for phpmyadmin in /config.json
2020-11-05T20:37:43+01 [D] Pulling phpmyadmin/phpmyadmin:latest for /phpmyadmin
2020-11-05T20:37:43+01 [I] Found new mariadb:latest image (sha256:1b3986d60f138825865a5c22e19d877f0318d60e88383a0e2fb8baa05bd5a461)
2020-11-05T20:37:13+01 [D] Got image name: mariadb:latest
2020-11-05T20:37:13+01 [D] No credentials for mariadb:latest in /config.json
2020-11-05T20:37:13+01 [D] Pulling mariadb:latest for /database
2020-11-05T20:37:07+01 [I] Found new gitea/gitea:latest image (sha256:e75238e09810f288ed353a76e22ffb2604c91a7ce4eb331472a8c7395b47da06)
2020-11-05T20:36:52+01 [D] Got image name: gitea/gitea:latest
2020-11-05T20:36:52+01 [D] No credentials for gitea in /config.json
2020-11-05T20:36:52+01 [D] Pulling gitea/gitea:latest for /gitea
2020-11-05T20:36:52+01 [D] Retrieving running containers
2020-11-05T20:36:52+01 [D] Checking containers for updated images
2020-11-05T20:36:52+01 [I] Running a one time update.
```

**Comment by piksel** [comment]
I might be wrong here, but I think your dependency is in the opposite direction. So it thinks that `gitea` is needed to start `database` and not the other way around.
Could you post your `docker-compose` (or equivalent) and the command line you use to start watchtower?

**Comment by Sveeeeeen** [comment]
> I might be wrong here, but I think your dependency is in the opposite direction. So it thinks that `gitea` is needed to start `database` and not the other way around.
> Could you post your `docker-compose` (or equivalent) and the command line you use to start watchtower?

I am running watchtower with WATCHTOWER_INCLUDE_STOPPED,WATCHTOWER_DEBUG,WATCHTOWER_RUN_ONCE, WATCHTOWER_CLEANUP parameter set to true. Nothing special at all.

**Comment by simonaronsson** [comment]
@Sveeeeeen Sorry, but we can't really help you without a redacted paste of your docker-compose file.

**Comment by Sveeeeeen** [comment]
> @Sveeeeeen Sorry, but we can't really help you without a redacted paste of your docker-compose file.

I do not use docker-compose.

**Comment by simonaronsson** [comment]
Then we need your `docker run` command for the containers in question as well as for watchtower.

**Comment by Sveeeeeen** [comment]
I told you the parameter already I am using

```
docker run -d \
--name watchtower \
-e TZ=Europe/Berlin \
-e WATCHTOWER_CLEANUP=true \
-e WATCHTOWER_INCLUDE_STOPPED=true \
-e WATCHTOWER_DEBUG=true \
-e WATCHTOWER_RUN_ONCE=true \
-v /var/run/docker.sock:/var/run/docker.sock \
containrrr/watchtower
```

**Comment by simonaronsson** [comment]
And for the containers that you're trying to add restarts with dependencies for? Come on, work with us a bit here if you want help. 😅

**Comment by pridit** [comment]
I am also getting a similar issue. I was under the impression that linked containers would not be an issue with watchtower, but maybe my understanding of this is flawed. I'm not a Docker expert so if this is an issue on my end please let me know. In specific I am using InfluxDB and Telegraf containers.

Here are my containers:
```
docker create \
  --name=influxdb \
  --restart unless-stopped \
  -p 8086:8086 \
  -v influxdb_data:/var/lib/influxdb \
  influxdb
```

```
docker create \
  --name=telegraf \
  --net=container:influxdb \
  --restart unless-stopped \
  -v /mnt/files:/data \
  telegraf
```

This is the watchtower output:
```
time="2021-02-10T22:45:01Z" level=info msg="Running a one time update."
time="2021-02-10T22:45:10Z" level=info msg="Found new telegraf:latest image (sha256:d780e0366b64214d3a2fdabf82f6f7a5ee9090caac1ece43d91e4f58f4935410)"
time="2021-02-10T22:45:12Z" level=info msg="Found new influxdb:latest image (sha256:c7870fcea362222d648815a18c6d949e3faae225b97b8c0ee7cc27b74e9d8c45)"
time="2021-02-10T22:45:16Z" level=info msg="Stopping /telegraf (dc7dd12cbf7e80fc04fd061a6f9dd0b2a84845bfc0e1518c4373a99b45739968) with SIGTERM"
time="2021-02-10T22:45:17Z" level=info msg="Stopping /influxdb (6a82cb0816a0d98ce3c92b7ab5fdd0b9877eb52a5733b623c0de0e1456f94da2) with SIGTERM"
time="2021-02-10T22:45:18Z" level=info msg="Creating /influxdb"
time="2021-02-10T22:45:19Z" level=info msg="Creating /telegraf"
time="2021-02-10T22:45:19Z" level=error msg="Error response from daemon: No such container: 6a82cb0816a0d98ce3c92b7ab5fdd0b9877eb52a5733b623c0de0e1456f94da2"
time="2021-02-10T22:45:19Z" level=info msg="Removing image sha256:d59d37a5129aa7d9a167c42790b6b55954e6891ee72818a7b58bb83b034674ba"
time="2021-02-10T22:45:20Z" level=info msg="Removing image sha256:2fe56be3a20fc85dc845c8c73e677bb9c51a4757cb632af00f18c63b27a3ce05"
```

**Comment by jubbens** [comment]
I've also had this issue. There are some containers that won't restart unless another container is already alive.

I think a reasonable solution would be an option to stop, update, and restart each container one at a time, instead of stopping all of them at once.

**Comment by simonaronsson** [comment]
We actually found a regression in the linked/depending containers logic that will be corrected with the `v1.3.0` release. Hopefully, this will solve your issues.

**Comment by simonaronsson** [comment]
Included in v1.4.0
